File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.cpp
// Last updated: 6/10/2026, 11:50:25 AM
class Solution
{
private:
// اللهم اهدنا يا رب اليك

public:
    int maxProfit(vector<int>& prices) {
        int mx = prices[prices.size() -1], ans = 0;   
        for(int i = prices.size() -2 ; i >= 0; i--){
            if(mx  > prices[i]) ans = max(ans , mx - prices[i]);
            else mx = prices[i];
        } 
        return ans;
    }
};

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 6/10/2026, 11:50:29 AM
import java.util.Scanner;
class Solution {
    public int maxProfit(int[] prices) {
    int size = prices.length , maX = 0 , j;    
    j = size - 1 ;                            
    for(int i = size-2 ; i >= 0 ; i--)
        {   if(prices[j] - prices[i] < 0)
                {j--; i++; continue;}
            if (maX < prices[j] - prices[i])
                maX = prices[j] - prices[i];
        }
        return(maX);
    }
}